kit/Kit.cpp

```diff
@@ -2425,7 +2425,11 @@ int pollCallback(void* pData, int timeoutUs)
     lock.unlock();
     if (v.empty())
     {
-        std::this_thread::sleep_for(std::chrono::microseconds(timeoutUs));
+        // On iOS, we cannot wait very long otherwise we are waiting here
+        // when the app opens a document on another thread.
+        // TODO: replace sleeping with a wait condition that can be
+        // interrupted if the user initiates opening a document.
+        std::this_thread::sleep_for(std::chrono::seconds(5));
     }
     else
     {
```

Comment:
The loop and kitPoll below is sometihng of an abomination. We should add a SocketPoll method for mobile - that can be passed this vector of SocketPolls - will suck all the FakeSocket fd's out of those - and then end up shoving them all into a int fakeSocketPoll(struct pollfd *pollfds, int nfds, int timeout) with a suitable timeout.
Then - inside that loop we can do this p->kitPoll(0); thing with a very short timeout to re-poll and dispatch the items per SocketPoll. Not beautiful, but far less problematic with multiple documents I think.